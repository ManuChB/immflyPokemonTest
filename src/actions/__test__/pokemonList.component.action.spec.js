import pokemonAction from '../pokemonList.component.action';

describe('PokemonList Actions', () => {
    it('initialize() should return action', () => {
        expect(pokemonAction.initialize()).toMatchSnapshot();
    });
    it('setPokemonList() should return action', () => {
        const pokemonList = [];
        expect(pokemonAction.setPokemonList(pokemonList)).toMatchSnapshot();
    });
    it('showPokemonInfo() should return action', () => {
        const pokemon = {
            name: 'eevee',
            url: '',
            height: 4,
            types: [],
            abilities: []
        };
        expect(pokemonAction.showPokemonInfo(pokemon)).toMatchSnapshot();
    });
    it('setSelectedPokemonInfo() should return action', () => {
        const pokemon ={
            name: 'eevee', 
            url: '', 
            height: 4, 
            types: [], 
            abilities: []
        };
        expect(pokemonAction.setSelectedPokemonInfo(pokemon)).toMatchSnapshot();
    });
    it('closeModal() should return action', () => {
        expect(pokemonAction.closeModal()).toMatchSnapshot();
    });
    it('setLoading() should return action', () => {
        const loading = false;
        expect(pokemonAction.setLoading(loading)).toMatchSnapshot();
    });
});
