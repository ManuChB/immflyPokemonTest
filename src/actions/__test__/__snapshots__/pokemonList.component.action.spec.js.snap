// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonList Actions closeModal() should return action 1`] = `
Object {
  "type": "CLOSE_MODAL",
}
`;

exports[`PokemonList Actions initialize() should return action 1`] = `
Object {
  "type": "INITIALIZE",
}
`;

exports[`PokemonList Actions setLoading() should return action 1`] = `
Object {
  "payload": false,
  "type": "LOADING",
}
`;

exports[`PokemonList Actions setPokemonList() should return action 1`] = `
Object {
  "payload": Array [],
  "type": "SET_POKEMON_LIST",
}
`;

exports[`PokemonList Actions setSelectedPokemonInfo() should return action 1`] = `
Object {
  "payload": Object {
    "abilities": Array [],
    "height": 4,
    "name": "eevee",
    "types": Array [],
    "url": "",
  },
  "type": "SET_SELECTED_POKEMON_INFO",
}
`;

exports[`PokemonList Actions showPokemonInfo() should return action 1`] = `
Object {
  "payload": Object {
    "abilities": Array [],
    "height": 4,
    "name": "eevee",
    "types": Array [],
    "url": "",
  },
  "type": "SHOW_POKEMON_INFO",
}
`;
